atomic_reactor/plugins/post_koji_import.py

```diff
@@ -442,38 +442,54 @@ def get_build(self):
 
         return build
 
-    def combine_metadata_fragments(self):
-        def add_buildroot_id(output, buildroot_id):
-            logfile, metadata = output
-            metadata.update({'buildroot_id': buildroot_id})
-            return Output(file=logfile, metadata=metadata)
+    def combine_metadata_fragments(self) -> Dict[str, Any]:
+        """Construct the CG metadata and collect the output files for upload later."""
+        def add_buildroot_id(output: Output, buildroot_id: str) -> Output:
+            output.metadata.update({'buildroot_id': buildroot_id})
+            return Output(filename=output.filename, metadata=output.metadata)
 
-        def add_log_type(output):
-            logfile, metadata = output
-            metadata.update({'type': 'log', 'arch': 'noarch'})
-            return Output(file=logfile, metadata=metadata)
+        def add_log_type(output: Output) -> Output:
+            output.metadata.update({'type': 'log', 'arch': 'noarch'})
+            return Output(filename=output.filename, metadata=output.metadata)
 
         try:
             self.pipeline_run_name = self.workflow.user_params['pipeline_run_name']
         except KeyError:
             self.log.error("No pipeline_run_name found")
             raise
 
-        metadata_version = 0
-
-        remote_source_file_outputs, kojifile_components = get_maven_metadata(self.workflow.data)
-
         build = self.get_build()
         buildroot = self.get_buildroot()
         buildroot_id = buildroot[0]['id']
+
+        # Collect the output files, which will be uploaded later.
+        koji_upload_files = self.workflow.data.koji_upload_files
+
+        output: List[Dict[str, Any]]  # List of metadatas
+        # The corresponding output file, only has one for source build
+        output_file: Optional[Output]
+
         output, output_file = self.get_output(buildroot_id)
+        if output_file:
+            koji_upload_files.append({
+                "local_filename": output_file.filename,
+                "dest_filename": output[0]["filename"],
+            })
+
+        # Collect log files
         osbs_logs = OSBSLogs(self.log, get_platforms(self.workflow.data))
-        output_files = [add_log_type(add_buildroot_id(md, buildroot_id))
-                        for md in osbs_logs.get_log_files(self.osbs, self.pipeline_run_name)]
+        output_log_files = [
+            add_log_type(add_buildroot_id(md, buildroot_id))
+            for md in osbs_logs.get_log_files(self.osbs, self.pipeline_run_name)
+        ]
+        for remote_source_file_output in output_log_files:
```

Comment:
Why is it called `remote_source_file_output`? They're just log files, aren't they? :thinking: